gen/common/rmw.ml

```diff
@@ -46,18 +64,18 @@ module
     let expand_rmw rmw = [rmw]
   end
 
-module LxSx(A:sig type arch_atom end) = struct
+module LxSx(A:RmwType) = struct
   include Make
     (struct
-      type atom = A.arch_atom
+      type arch_atom = A.arch_atom
```

Comment:
Why `arch_atom` instead of `atom`?

Reply:
To be honest I cannot remember too much now. There are some part of the `diy` code use `arch_atom` which means to refer those annotations/atoms that are specifically to an architecture. But I think we do not need to use different name. I will systematically review at least for the part of code I changed and use `atom` consistently.

Reply:
Thank you!